Layout external providers so it works locally

astefanutti opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

Setup external providers so it works locally OOTB.

## How Has This Been Tested?

LLS server successfully starts with KFTO remote provider.

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

leseb reviewed (changes requested):
I'm about to update the working branch, we need a new top level tree that represents the provider type.

So you need:

```
providers.d/remote/post_training/instructlab_kft.yaml
```

astefanutti commented:
@leseb thanks for the heads-up. I've rebased on your latest changes and added the `remote` segment.

Successfully tested:

```
INFO     2025-03-27 13:51:38,339 llama_stack.distribution.distribution:164 core: Loading remote provider spec from      
         /Users/antonin/Development/opendatahub-io/llama-stack-provider-kft/providers.d/remote/post_training/instructlab
         _kft.yaml                                                                                                      
INFO     2025-03-27 13:51:38,340 llama_stack.distribution.distribution:177 core: Loaded remote provider spec for        
         remote::instructlab_kft from                                                                                   
         /Users/antonin/Development/opendatahub-io/llama-stack-provider-kft/providers.d/remote/post_training/instructlab
         _kft.yaml
```

leseb commented on pyproject.toml:
```diff
@@ -1,6 +1,6 @@
 [project]
 dependencies = [
-    "llama-stack @ git+https://github.com/cdoern/llama-stack.git@feat-658",
+    "llama-stack @ git+https://github.com/astefanutti/llama-stack.git@feat-658-1238",
```
Why are we making this change? Why don't we use the https://github.com/meta-llama/llama-stack/pull/1672

leseb replied:
nvm we need the job/async code too.
Just curious why not use Charlie branche anymore?

astefanutti replied:
Right https://github.com/booxter/llama-stack/tree/scheduler-for-torchtune is needed.

@cdoern branch isn't rebased on top of your latest changes.
